.github/workflows/release.yaml

```diff
@@ -134,11 +142,18 @@ jobs:
             echo "release_title=${VERSION_OR_SHA}" >> $GITHUB_OUTPUT
           fi
 
-      - name: Download artifacts
-        uses: actions/download-artifact@v4
-        with:
-          pattern: release*
-          path: releases
+      - name: Check release version
+        if: github.ref_type == 'tag'
+        run: |
+          TAG="${{ steps.release.outputs.version_or_sha }}"
+          CARGO_PACKAGE_VERSION="$(cargo pkgid | cut -d "#" -f2)"
+          CHANGELOG_VERSION="${{ steps.changelog_reader.outputs.version }}"
+          # All versions must be equal
+          if [ "${CARGO_PACKAGE_VERSION}" != "${TAG}" ] || [ "${CARGO_PACKAGE_VERSION}" != "${CHANGELOG_VERSION}" ]; then

```

Comment:
I guess it only checks if the current Cargo.toml version matches the last changelog version, does it?

What would be the easiest way for things to break? - Just a little brainstorm before we can merge.

Reply:
Here is the test for `zksolc`:
https://github.com/matter-labs/era-compiler-solidity/actions/runs/9839561820/job/27162678159

```
Version mismatch: TAG=1.5.3, CARGO_PACKAGE_VERSION=1.5.2, CHANGELOG_VERSION=1.5.1
Please update the version in Cargo.toml, CHANGELOG.md and tag the commit with the same version.
```

For `zkvyper` the code is the same, so I believe we should be good to go.